File: src/app/models/ShiftWorkerModel.ts
class ShiftWorkerModel extends WorkerModel {
    workingHoursPerDate:Map<Date,number>

    constructor()
    constructor(id:number,name:string,workingHoursPerDate:Map<Date,number>,)
    constructor(id?:number,name?:string,workingHoursPerDate?:Map<Date,number>,) {
        super(id ?? 0, name ?? ""); 
        this.id = id ?? 0
        this.name = name ?? "";
        this.workingHoursPerDate = workingHoursPerDate ?? new Map<Date,number>;
      }
}

File: src-tauri/src/services/models/shift_worker_model.rs
use std::collections::HashMap;

use serde::{Deserialize, Serialize};

use super::worker_model::WorkerModel;
#[derive(Debug, Deserialize,Serialize)]
pub struct ShiftWorkerModel{
    pub parent:WorkerModel,
    pub working_hours_per_date:HashMap<u32, u32>
} 

impl ShiftWorkerModel {
    pub fn new(id:u32,name:String,working_hours_per_date:HashMap<u32, u32>) -> Self {
        Self {
            parent:WorkerModel::new(id, name),
            working_hours_per_date:working_hours_per_date
        }
    }
}